discovery/azure/azure.go

```diff
@@ -126,6 +130,12 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return errors.Errorf("unknown authentication_type %q. Supported types are %q or %q", c.AuthenticationMethod, authMethodOAuth, authMethodManagedIdentity)
 	}
 
+	// Because Azure's mandatory authentication methods use the authorization header, others can't be configured.
+	if c.HTTPClientConfig.BasicAuth != nil || c.HTTPClientConfig.Authorization != nil || c.HTTPClientConfig.OAuth2 != nil ||
+		len(c.HTTPClientConfig.BearerToken) > 0 || len(c.HTTPClientConfig.BearerTokenFile) > 0 {
+		return errors.New("standard authentication methods cannot be used, only azure OAuth or ManagedIdentity")
```

Comment:
Not sure if this error message is clear enough

Reply:
we should not do this, instead just let azure calls fail

Reply:
I disagree. The authentication methods _will_ override the Azure ones, and the request _will_ fail. It’s pretty much certain (I think). So why let users debug confusing request failures when we can just tell them explicitly on startup?